StitchVolumes/StitchVolumes.py

```diff
@@ -0,0 +1,706 @@
+        # - the inner corner voxel values (in the overlap region) should not
+
+        self.delayDisplay("Test passed")
+
+
+####################
+#
+# Subfunctions
+#
+####################
+
+
+def get_RAS_center(vol_node):
+    """Find the RAS coordinate center of the image volume from the RAS bounds"""
+    b = [0] * 6
+    vol_node.GetBounds(b)
+    cen = [np.mean([b[0], b[1]]), np.mean([b[2], b[3]]), np.mean([b[4], b[5]])]
+    return cen
+
+
+def find_dim_to_stitch(orig_nodes, resamp_node):
+    # This function determines the dimension to stitch the original nodes along by
+    # finding the image axis dimension (I,J,or K) which is best aligned with the
+    # vector between the centers of the furthest apart original volumes.
+    # A resampled volume is needed just in case its IJK direction matrix
+    # differs from the original nodes. I believe this method should be
+    # fairly robust.
+    RAS_centers = [get_RAS_center(vol) for vol in orig_nodes]
+    dists = [
+        np.linalg.norm(np.subtract(RAS_center, RAS_centers[0]))
+        for RAS_center in RAS_centers
+    ]
+    furthest_from_first = np.argmax(dists)
+    stitch_vect = np.subtract(RAS_centers[0], RAS_centers[furthest_from_first])
+    stitch_vect = stitch_vect / np.linalg.norm(stitch_vect)
+    # RAS_biggest_change_idx= np.argmax(np.abs(stitch_vect))
+    # Now I need to know which image volume axis (I,J,or K) is most aligned with the stitching vector
+    # We can do this by comparing the dot products of each of the I J and K vectors with the stitch
+    # vector.  The one with the maximum abs dot product is the winner
+    ijkdirs = [
+        [0, 0, 0],
+        [0, 0, 0],
+        [0, 0, 0],
+    ]  # NOTE these will be the ROWS, not columns of ijk to ras matrix
+    resamp_node.GetIJKToRASDirections(ijkdirs)  # fill in values
+    ijkdirs_np = np.array(ijkdirs)
+    # Compute dot products with the columns of ijk to ras matrix
+    absDotsIJK = [np.abs(np.dot(d, stitch_vect)) for d in ijkdirs_np.T]
+    IJKmatchIdx = np.argmax(absDotsIJK)
+    KJImatchIdx = 2 - IJKmatchIdx
+    dim_to_stitch = KJImatchIdx
+    return dim_to_stitch
+
+
+def createOrReplaceNode(name, nodeClass="vtkMRMLScalarVolumeNode"):
+    try:
+        node = slicer.util.getNode(name)
+    except:
+        node = slicer.mrmlScene.AddNewNodeByClass(nodeClass, name)
+    return node
+
+
+def resample_volume(roi_node, input_vol_node, output_vol_name):
+    """Carry out the cropping of input_vol_node to the space described by roi_node"""
+    cropVolumeNode = slicer.vtkMRMLCropVolumeParametersNode()
+    cropVolumeNode.SetScene(slicer.mrmlScene)
+    cropVolumeNode.SetName("MyCropVolumeParametersNode")
+    cropVolumeNode.SetIsotropicResampling(False)
+    cropVolumeNode.SetInterpolationMode(
+        cropVolumeNode.InterpolationNearestNeighbor
+    )  # use nearest neighbor to avoid resampling artifacts
+    cropVolumeNode.SetFillValue(
+        0
```

Comment:
I would recommend to compute fill value by vtkMRMLScalarVolumeNode::GetImageBackgroundScalarComponentAsDouble() (from the median of the 8 corner voxels of the volume)

Reply:
I use the fill value being zero somewhere else in the logic, and I will need to fix that dependence before changing this.  I agree that it makes sense to base the value on the image corners rather than force to zero for all images.